tests/pruners_tests/test_hyperband.py

```diff
@@ -216,3 +216,27 @@ def objective(trial: optuna.trial.Trial) -> float:
 def test_incompatibility_between_bootstrap_count_and_auto_max_resource() -> None:
     with pytest.raises(ValueError):
         optuna.pruners.HyperbandPruner(max_resource="auto", bootstrap_count=1)
+
+
+def test_hyperband_pruner_and_grid_sampler() -> None:
+    pruner = optuna.pruners.HyperbandPruner(
+        min_resource=MIN_RESOURCE, max_resource=MAX_RESOURCE, reduction_factor=REDUCTION_FACTOR
+    )
+
+    search_space = {"x": [-50, 0, 50], "y": [-99, 0, 99]}
+    sampler = optuna.samplers.GridSampler(search_space)
+
+    study = optuna.study.create_study(sampler=sampler, pruner=pruner)
+
+    def objective(trial: optuna.trial.Trial) -> float:
+        for i in range(N_REPORTS):
+            trial.report(i, step=i)
+
+        x = trial.suggest_uniform("x", -100, 100)
```

Comment:
A nit, but we usually use `suggest_float` instead of `suggest_uniform` for newly added code. (c.f., #2344 and #1470. I couldn't find this convention in the [wiki](https://github.com/optuna/optuna/wiki/Coding-Style-Conventions), but I think it should be there.)
Could you use `suggest_float`, please?
```suggestion
        x = trial.suggest_float("x", -100, 100)
```